tests/tensorbox/test_autograd_box.py

```diff
@@ -0,0 +1,194 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Unit tests for AutogradBox subclass"""
+import pytest
+
+autograd = pytest.importorskip("autograd")
+
+import pennylane as qml
+from pennylane import numpy as np
+from pennylane.tensorbox.autograd_box import AutogradBox
+
+
+def test_creation():
+    """Test that a AutogradBox is automatically created from a PennyLane numpy tensor"""
+    x = np.array([0.1, 0.2, 0.3])
+    res = qml.tensorbox.TensorBox(x)
+    assert isinstance(res, AutogradBox)
+    assert res.interface == "autograd"
+    assert isinstance(res.unbox(), np.ndarray)
+    assert np.all(res == x)
+
+
+def test_astensor_list():
+    """Test conversion of a list to PennyLane tensors"""
+    x = np.array([0.1, 0.2, 0.3])
+    y = [0.4, 0.5, 0.6]
+
+    res = qml.tensorbox.TensorBox(x).astensor(y)
+    assert isinstance(res, np.tensor)
+    assert np.all(res == y)
+
+
+def test_astensor_array():
+    """Test conversion of numpy arrays to PennyLane tensors"""
+    x = np.array([0.1, 0.2, 0.3])
+    y = np.array([0.4, 0.5, 0.6])
+
+    res = qml.tensorbox.TensorBox(x).astensor(y)
+    assert isinstance(res, np.tensor)
+    assert np.all(res == y)
+
+
+def test_cast():
+    """Test that arrays can be cast to different dtypes"""
+    x = np.array([1, 2, 3])
+
+    res = qml.tensorbox.TensorBox(x).cast(np.float64)
+    expected = np.array([1.0, 2.0, 3.0])
+    assert np.all(res == expected)

```

Comment:
interesting, i did not know `==` picks up the type

Reply:
wouldn't it be safer to check the dtype of the elements in `res`?

Reply:
Added!